zaza/openstack/utilities/openstack.py

```diff
@@ -2651,10 +2653,68 @@ def upload_image_to_glance(glance, local_path, image_name, disk_format='qcow2',
     return image
 
 
+def is_ceph_image_backend(model_name=None):
+    """Check if glance is related to ceph, therefore using ceph as backend.
+
+    :returns: True if glance is related to Ceph, otherwise False
+    :rtype: bool
+    """
+    status = juju_utils.get_full_juju_status(model_name=model_name)
+    result = False
+    try:
+        result = 'ceph-mon' in (
+            status['applications']['glance']['relations']['ceph'])
+    except KeyError:
+        pass
+    if result is False:
+        logging.info("Did not detect Ceph related to Glance")
+    return result
+
+
+def ensure_image_format(local_path, disk_format):
+    """Convert the image if its format differs from the desired format.
```

Comment:
This isn't quite right. What the function does is "if the disk_format param is 'raw' and the image format is 'qcow2' then convert the image to 'raw', otherwise don't convert it.  And if there are any errors, don't convert it, and return the original path."

So, for example, the function would be more efficient if it just bailed out on checking for the disk_format being "raw".  But then, the question becomes, "why even call the function if the `disk_format` isn't 'raw'"?  i.e. the caller can decide not to call the function if the required disk_format isn't 'raw'.

So taking all this into account, I think:

* the function should be called "convert_image_format_to_raw_if_qcow2()".  That's what it *does*.
* Don't call the function if disk_format isn't going to 'raw'.
* Don't pass 'disk_format' to the function; it's only purpose is to check if it's 'raw' or do nothing.
* At the call site, only call the function if the disk_format is 'raw'.

Reply:
done